core/src/main/java/org/mule/runtime/core/api/construct/FlowBuilder.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.api.construct;
+
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.exception.MessagingExceptionHandler;
+import org.mule.runtime.core.api.processor.Processor;
+import org.mule.runtime.core.api.processor.strategy.ProcessingStrategyFactory;
+import org.mule.runtime.core.api.source.MessageSource;
+import org.mule.runtime.core.internal.construct.DefaultFlow;
+
+import java.util.List;
+
+/**
+ * Creates instances of {@link Flow} with a default implementation
+ * 
+ * <p/>
+ * Builder instance can be configured using the methods that follow the builder pattern until the flow is built. After that
+ * point, builder methods will fail to update the builder state.
+ */
+public class FlowBuilder {
+
+  private final String name;
+  private final MuleContext muleContext;
+  private MessageSource messageSource;
+  private List<Processor> messageProcessors;
+  private MessagingExceptionHandler exceptionListener;
+  private ProcessingStrategyFactory processingStrategyFactory;
+  private DefaultFlow flow;
+
+  /**
+   * Creates a new builder
+   *
+   * @param name name of the flow to be created. Non empty.
+   * @param muleContext context where the flow will be registered. Non null.
+   */
+  public FlowBuilder(String name, MuleContext muleContext) {
```

Comment:
Can we use the same pattern as with Event/Message and others, where you create builder from `Flow.builder()` or `Flow.builder(name, muleContet)`?